rcl/test/rcl/test_wait.cpp

```diff
@@ -41,3 +46,132 @@ TEST(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), test_resize_to_zero) {
   ret = rcl_wait_set_fini(&wait_set);
   ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
 }
+
+// Some test cases for the waitset
+TEST(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), finite_timeout) {
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rcl_ret_t ret = rcl_wait_set_init(&wait_set, 0, 0, 1, 0, 0, rcl_get_default_allocator());
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+
+  int64_t timeout = 1000000;  // nanoseconds
+  std::chrono::steady_clock::time_point before_sc = std::chrono::steady_clock::now();
+  ret = rcl_wait(&wait_set, timeout);
+  std::chrono::steady_clock::time_point after_sc = std::chrono::steady_clock::now();
+  // Check time
+  int64_t diff = (after_sc - before_sc).count();
```

Comment:
Are you sure `.count` will always return nanoseconds? I think it might depend on the default storage of the clock being used (which might be implementation specific). I'd recommend doing an explicit `duration_cast` to nanoseconds first.